src/python_testing/TC_DeviceConformance.py

```diff
@@ -393,12 +393,45 @@ def record_warning(location, problem):
                     if conformance_decision_with_choice.decision == ConformanceDecision.MANDATORY and cluster_id not in server_clusters:
                         record_error(location=location,
                                      problem=f"Mandatory cluster {cluster_requirement.name} for device type {xml_device.name} is not present in the server list")
-                        success = False
+                        continue
 
                     if cluster_id in server_clusters and not conformance_allowed(conformance_decision_with_choice, allow_provisional):
                         record_error(location=location,
                                      problem=f"Disallowed cluster {cluster_requirement.name} found in server list for device type {xml_device.name}")
-                        success = False
+                        continue
+
+                    if cluster_id not in server_clusters:
+                        # Optional cluster not on this endpoint
+                        continue
+
+                    cluster = Clusters.ClusterObjects.ALL_CLUSTERS[cluster_id]
+                    feature_map = endpoint[cluster][cluster.Attributes.FeatureMap]
+                    attribute_list = endpoint[cluster][cluster.Attributes.AttributeList]
+                    cmd_list = endpoint[cluster][cluster.Attributes.AcceptedCommandList]
+                    for mask, conformance in cluster_requirement.feature_overrides.items():
+                        conformance_decision_with_choice = conformance(feature_map, attribute_list, cmd_list)
+                        if conformance_decision_with_choice.decision == ConformanceDecision.MANDATORY and (feature_map & mask == 0):
+                            record_error(
+                                location=location, problem=f"Feature mask {mask} in cluster {cluster_requirement.name} is required by element override for device type {xml_device.name}, but is not present in the feature map")
+                        if not conformance_allowed(conformance_decision_with_choice, allow_provisional) and (feature_map & mask != 0):
+                            record_error(
+                                location=location, problem=f"Feature mask {mask} in cluster {cluster_requirement.name} is disallowed by element override for device type {xml_device.name}, but is present in the feature map")
+                    for id, conformance in cluster_requirement.attribute_overrides.items():
+                        conformance_decision_with_choice = conformance(feature_map, attribute_list, cmd_list)
+                        if conformance_decision_with_choice.decision == ConformanceDecision.MANDATORY and id not in attribute_list:
```

Comment:
This coould be a method of ConformanceDecision as it's repeated a lot.

Also, the 3 for-loops could be split in functions taking a couple attributes so the test logic is more obvious that it's:

- Check feature overrides
- Check attribute overrides
- Check command overrides